ui/src/main/java/edu/wpi/grip/ui/codeGeneration/TemplateMethods.java

```diff
@@ -0,0 +1,119 @@
+  }
+
+  public static String parseSocketValue(Socket socket) {
+    String value = socket.getValue().toString();
+    if (socket.getSocketHint().getView().equals(SocketHint.View.NONE)) {
+      return "null" + value;
+    } else if (value.contains("Optional[")) {
+      return value.substring(value.indexOf("[") + 1, value.lastIndexOf("]"));
+    } else {
+      return value;
+    }
+  }
+
+  public static String parseSocketName(Socket socket) {
+    String name = socket.getSocketHint().getIdentifier();
+    return CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, name.replaceAll("\\s", ""));
+  }
+
+  public static String parseSocketType(Socket socket) {
+    String type = socket.getSocketHint().getType().getSimpleName();
+    if (socket.getSocketHint().getView().equals(SocketHint.View.SELECT)) {
+      if (BorderTypesEnum.class.equals(socket.getSocketHint().getType()) || CmpTypesEnum.class
+          .equals(socket.getSocketHint().getType())) {
+        type += "CoreEnum";
+      }
+    } else if (type.equals("ContoursReport")) {
+      type = "ArrayList<MatOfPoint>";
+    } else if (type.equals("BlobsReport")) {
+      type = "MatOfKeyPoint";
+    } else if (type.equals("LinesReport")) {
+      type = "ArrayList<Line>";
+    }
+    return type;
+  }
+
+  public static String opName(String name) {
+    if (name.contains("CV ")) {
+      return name.substring(3);
+    } else return name;
+  }
+
+  public static String javaName(String name) {
+    return CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, name.replaceAll("\\s", ""));
+  }
+
+  public static String callJavaOp(TStep step) {
+    String num = "S" + step.num();
+    StringBuilder out = new StringBuilder();
+    out.append(javaName(step.name()));
+    out.append("(");
+    for (TInput input : step.getInputs()) {
+      out.append(input.name());
+      out.append(num);
+      out.append(", ");
+    }
+    if (step.name().equals("Threshold_Moving")) {
+      out.append("this.lastImage");
+      out.append(num);
+      out.append(", ");
+    }
+    if (!step.getOutputs().isEmpty()) {
+      for (int i = 0; i < step.getOutputs().size() - 1; i++) {
+        out.append(step.getOutputs().get(i).name());
+        out.append(", ");
+      }
+      out.append(step.getOutputs().get(step.getOutputs().size() - 1).name());
+    }
+    out.append(")");
+    return out.toString();
+  }
+
+  public static String cName(String name) {
+    // name is something like "CV_medianBlur" or "Find_Contours"
+    if (name.startsWith("CV_")) {
+      // OpenCV operation
+      String op = name.replaceFirst("CV_", "");
+      if (op.contains("_")) {
+        return "CV" + CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, op.toLowerCase());
+      } else {
+        return "CV" + CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, op);
+      }
+    } else {
+      // GRIP operation
+      return CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, name.toUpperCase());
+    }
+  }
```

Comment:
It would be nice if we could move the definition of 'callJavaOp' and 'cName' into templates themselves for each language that uses them. That would help make this functionality more extensible.